Allow users to provide a merchant account per store

luukveenis opened the pull request:
It's possible that users may want transactions for different stores to be processed by different merchant accounts in Adyen. A typical use case would be that the user has stores for different countries and they have different bank accounts for each country, which are linked to different merchant accounts in Adyen.

Unfortunately we don't have easy access to the store from inside our gateway actions. The only thing they all share is the `psp_reference` which is recorded on the payment in the `response_code` field. In the case of `cancel`, this is the only parameter it receives, so looking up the store using the response code was the only way I could think to make this work for all actions.

adammathys reviewed (approved):
:+1: Seems reasonable to me.

Dkendal commented on app/models/concerns/spree/gateway/adyen_gateway.rb:
```diff
@@ -64,9 +66,14 @@ def handle_response(response, original_reference = nil)
       )
     end
 
+    def store_for_payment(psp_reference)
```
It's pretty difficult to understand how this is intended to work, even after a couple passes. 

After about 10 minutes I now understand this is just finding the store that had a payment with this psp ref.

It would be great to see more explicit documentation as to the intent of this implementation, and why it's required.

In my opinion this is complex enough, and has enough independent concerns from this module that it should be moved into it's own class/module.

Dkendal replied:
To clarify: I'm not referring to just this method but the whole periphery of it.

Commit:
Allow users to provide a merchant account per store

For applications with multiple stores, it's possible that users may want
transactions for different stores to be processed by different merchant
accounts in Adyen. A typical use case would be that the user has stores
for different countries and they have different bank accounts for each
country. Usually Adyen will link the bank accounts to different merchant
accounts and leave it up to us to pass the correct merchant account with
the payment request.

We don't always have access to the store when we're creating payment
requests, so I needed a way to look it up given the limited information
we have. In some cases, such as the `cancel` gateway action, the only
information we have is the psp reference for the payment. The
AccountLocator module provides several methods to look up the merchant
account when we don't have access to the store directly.

Dkendal reviewed (approved):
This looks way better, thanks @luukveenis
